solutions/0900-0999/0912-sort-an-array-medium.md

```diff
@@ -173,4 +173,155 @@ void quickSort(vector<int>& nums, int l, int r) {
 }
 ```
 </TabItem>
+</Tabs>
+
+## Approach 3: Merge Sort
+
+Standard merge sort algorithm, following a divide and conquer approach, division of sub-array by mid values till sub-array length
+reaches one, then merging parts by either out-place or in-place merging in either ascending or descending order. 
+
+### Merge Sort with Outplace Merging
+
+In this approach, merging happen in `seperate array` of size, then it is passed on in the above function calls.
+
+In mergeSort() function:
+    Dividing the array into 2 subarrays(one is $left$ and other is $right$) till their length reaches one(because no futher division can take place), then passing left and right parts into merge() function for sorting in correct order.
+
+In merge() function:
+    We merge the two subarrays( $left$ and $right$ ) into another array $mix$ . This merge operation stores elements from 
+    $left$ and $right$ in an ascending order, by comparing elements from individual subarrays.
+    Finally, since one subarray would have greater elements then the other one, so we add those elements seperately.
+    Then, we return the sorted subarray into the above function call.
```

Comment:
please run the app locally to verify. the content is all packed which is quite hard to read.